### ISSUE TITLE
Define Rules which apply to nodes or edges individually

### ISSUE THREAD

**Issue by spflueger** [ComPWA/expertsystem#303]
Some rules like the Gellmann Nishijima Rule are subject only to edge properties. The new spin validity rules fall under this category as well. Rules that only work with node properties or edge properties are needed. This simplifies and speeds up the code.

Also, these rules have to be optional in the sense that if no isospin is defined, then also the isospin validity check has to pass.

- [x] Create two new rule Protocols.
- [x] Change rules that only require edge or node quantum numbers to satisfy the new Protocols.
- [x] Adapt solving code to the new edge and node only rules.
- [x] ~~Replace the Gellmann Nishijima checking code in Particle to use the new rule instead.~~

Unfortunately with the solving interface using `Particle` introduces a cyclic dependency when trying to use the Gellmann Nishijima rule in the Particle initialization. This can be fixed after #369. Moving this task into a new issue.

**Comment by spflueger** [comment]
With PR #331 this issue can be handled more easily now. The idea is to introduce two new `Protocols` that allow Rules with single arguments of type `Edge-` and `NodeQuantumNumbers`, and adapt the solving code to link edges and nodes individually to these rules.
